SharedPackages/DataBrokerProtectionCore/Sources/DataBrokerProtectionCore/CCF/ActionsHandler.swift

```diff
@@ -41,16 +43,25 @@ public class ActionsHandler {
         guard let lastExecutedActionIndex = self.lastExecutedActionIndex else {
             // If last executed action index is nil. Means we didn't execute any action, so we return the first action.
             self.lastExecutedActionIndex = 0
-            return step.actions.first
+            return actions.first
         }
 
         let nextActionIndex = lastExecutedActionIndex + 1
 
-        if nextActionIndex < step.actions.count {
+        if nextActionIndex < actions.count {
             self.lastExecutedActionIndex = nextActionIndex
-            return step.actions[nextActionIndex]
+            return actions[nextActionIndex]
         } else {
             return nil // No more actions to execute
         }
     }
+
+    public func insert(actions: [Action]) {
+        if let lastExecutedActionIndex {
+            self.actions.insert(contentsOf: actions, at: lastExecutedActionIndex + 1)
```

Comment:
An extra boundary check for `lastExecutedActionIndex + 1` would be nice.

Reply:
Done, added a check for this as part of the `if let`, falling back on the `append` if it's not true